agentlib/core/environment.py

```diff
@@ -43,137 +40,230 @@ class EnvironmentConfig(BaseModel):
         """Get the current time of the environment.
         If RT is enabled, the unix-time is returned."""
-        if self.config.rt:
-            return self.now + self.t_start + self.offset
-        # Else return
-        return self.now + self.offset
+        return time.time() + self.config.offset
 
-    def clock(self):
-        """Define a clock loop to increase the now-timer every other milisecond
-        (Or whatever t_sample is)"""
+    def pretty_time(self) -> str:
+        """Returns the time in a nice format. Datetime if realtime, seconds if not
+        realtime. Implemented as rt_time or sim_time"""
+        return datetime.fromtimestamp(self.time).strftime("%d-%b-%Y %H:%M:%S")
+
+    def silent_clock(self):
+        """A silent clock, which does not log anything."""
         # if log level is not info or debug, this process can terminate
         while True:
-            logger.info("Current simulation time: %s", self.pretty_time())
             yield self.timeout(self.config.t_sample)
 
+
+class ScaledRealtimeEnvironment(simpy.RealtimeEnvironment, CustomSimpyEnvironment):
+    """A customized version of the simpy environment. Handles execution of modules
+    processes and manages time for scaled real time execution mode."""
+
+    def __init__(self, *, config: EnvironmentConfig):
+        super().__init__(
+            initial_time=config.offset, factor=config.factor, strict=config.strict
+        )
+        self._config = config
+        if self.config.clock:
+            self.process(self.clock())
+        else:
+            self.process(self.silent_clock())
+
+    def run(self, until: Optional[Union[SimTime, Event]] = None) -> Optional[Any]:
+        self.sync()
+        return super().run(until=until)
+
+    @property
+    def time(self) -> float:
+        """Get the current time of the environment.
+        If RT is enabled, the unix-time is returned."""
+        return self.now
+
     def pretty_time(self) -> str:
         """Returns the time in a nice format. Datetime if realtime, seconds if not
         realtime. Implemented as rt_time or sim_time"""
-        if self.config.rt:
-            return self.rt_time()
-        return self.sim_time()
-
-    def rt_time(self) -> str:
-        """Returns the current real time in datetime format, based on unix timestamp."""
-        try:
-            return datetime.fromtimestamp(self.time).strftime("%d-%b-%Y %H:%M:%S")
-        except TypeError:
-            # when the rt environment has not yet started, we cannot have a time
-            return ""
-
-    def sim_time(self) -> str:
-        """Returns the current simulation time in seconds, with two decimals of
-        precision."""
         return f"{self.time:.2f}s"
+
+    def silent_clock(self):
+        """A silent clock, which does not log anything."""
+        # if log level is not info or debug, this process can terminate
+        while True:
+            yield self.timeout(self.config.t_sample)
```

Comment:
In the monkey-case patch, you definitly need to add the License of simpy to the file and possibly the agentlib.

Reply:
I added the license to the docstring of the monkey patch. 
Maybe we should have a call about this one?